apps/projects/app/components/Panel/NewProject/NewProject.js

```diff
@@ -113,110 +121,187 @@ const NewProject = () => {
     repoArray: PropTypes.array.isRequired,
   }
 
-  return (
-    <React.Fragment>
-      <div css={`margin-top: ${3 * GU}px`}>
-        <Text weight="bold">
-            Which repos do you want to add?
-        </Text>
-        <div>
-          <Query
-            fetchPolicy="cache-first"
-            query={GET_REPOSITORIES}
-            onError={console.error}
-          >
-            {({ data, loading, error, refetch }) => {
-              if (data && data.viewer) {
-
-                const reposDownloaded = filterAlreadyAdded(data.viewer.repositories.edges)
-
-                const visibleRepos = filter ? filterByName(reposDownloaded) : reposDownloaded
-
-
-                const repoArray = visibleRepos.map(repo => ({
-                  title: repo.node.nameWithOwner,
-                  description: '',
-                  node: repo.node,
-                }))
-
-                return (
-                  <div>
-                    <TextInput
-                      type="search"
-                      style={{ margin: '16px 0', flexShrink: '0' }}
-                      placeholder="Search"
-                      wide
-                      value={filter}
-                      onChange={updateFilter}
-                      adornment={
-                        filter === '' && (
-                          <IconSearch
-                            css={`
-                                color: ${theme.surfaceOpened};
-                                margin-right: 8px;
-                              `}
-                          />
-                        )
-                      }
-                      adornmentPosition="end"
-                      ref={searchRef}
-                      aria-label="Search"
+  const RepoQuery = () => (
```

Comment:
Why define this inside `GithubRepoList` instead of as its own standalone component?

Reply:
It was one cleanup I wasn't convinced needed to be done immediately - but you're right, I've split it into 3 components.